### PR TITLE
Fix one cause of duplicate points.

### PR DESCRIPTION
**Pull request by zerebubuth**
After merging, simplify with a very small tolerance to remove duplicate and almost-colinear points.

Connects to mapzen/vector-datasource#569.

@rmarianski could you review, please?

### CODE REVIEW

**Comment by zerebubuth** [comment]
Connects to mapzen/vector-datasource#569.

File: TileStache/Goodies/VecTiles/transform.py [hunk 1]
```diff
@@ -1702,16 +1704,31 @@ def _linemerge(geom):
...
+        # almost duplicate or nearly colinear points can occur due to
+        # numerical round-off or precision in the intersection algorithm, and
+        # this should help get rid of those. see also:
+        # http://lists.gispython.org/pipermail/community/2014-January/003236.html
+        #
+        # the tolerance here is hard-coded to a fraction of the coordinate
+        # magnitude. there isn't a perfect way to figure out what this tolerance
+        # should be, so this may require some tweaking.
+        epsilon = max(map(abs, result_geom.bounds)) * float_info.epsilon * 1000
+        result_geom = result_geom.simplify(epsilon, True)
```
Comment (rmarianski) [inline TileStache/Goodies/VecTiles/transform.py:1729]: I take it this is helping, but out of curiosity wouldn't the simplification that occurs on the geometry itself afterwards based on zoom level take care of this? Or is this simplification more aggressive here?

**Comment by rmarianski** [comment]
:+1:

File: TileStache/Goodies/VecTiles/transform.py (hunk 1 above)
Comment (zerebubuth) [inline TileStache/Goodies/VecTiles/transform.py:1729]: We simplify until zoom 16, so the later simplification step wouldn't (and didn't) fix this at zoom 16. It seemed like this was worth having as the simplification step seems like it ought to be conceptually separate from the geometry merging, and that either should be independently able to produce valid results.

The simplification is about as un-aggressive as I could make it - on a mercator coordinate, it shouldn't ever be bigger than `4.45e-6`, and it's possible we could dial down that fixed factor of `1000` to something smaller. We'd just have to be careful to not re-introduce the duplicate points.

Of course, at this stage they're not _actually_ duplicates - they differed by `1.0e-9` - but that's close enough when we come to encode in MVT or GeoJSON. (But apparently not TopoJSON!)

**Pull request merged**